### PR TITLE
feat(ext.commands): add LargeRange

### LINKED ISSUE #787: combine commands.LargeInt with commands.Range
### Summary

LargeInt cannot be restricted like Range, and Range cannot be a big int.

### What is the feature request for?

disnake.ext.commands

### The Problem

LargeInt is an integer bigger than 2 ** 64, and Range *only* works on numbers below 2 ** 64. It would be nice to have a combination of the two, possibly even merge them into one class.

We also don't have LargeFloat right now, so a modified Range could potentially support that.

### The Ideal Solution

It would be nice to have a combination of the two, possibly even merge them into one class.

We also don't have LargeFloat right now, so a modified Range could potentially support that.

### The Current Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by EmreTech**
## Summary

This PR adds support for large numbers (numbers bigger than 2 ** 53 or smaller than -2 ** 53) in a version of `commands.Range` named `commands.LargeRange`.

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: changelog/787.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add large numbers support to :class:`commands.Range` and add :class:`commands.LargeFloat`.
```
Comment (shiftinv) [inline changelog/787.feature.rst:1]:
```suggestion
|commands| Add large numbers support to :class:`~ext.commands.Range` and add :class:`~ext.commands.LargeFloat`.
```

File: disnake/ext/commands/params.py [hunk 2]
```diff
@@ -680,18 +710,27 @@ def parse_annotation(self, annotation: Any, converter_mode: bool = False) -> boo
             self.min_value = annotation.min_value
             self.max_value = annotation.max_value
             annotation = annotation.underlying_type
+
+            if self.min_value < -(2**53) or self.max_value > 2**53:
+                self.large = True
```
[the remark this answers is not in the document]
Comment (EmreTech) [reply disnake/ext/commands/params.py:715]: Wdym by "if something doesn't check out"?

[the remark this answers is not in the document]
Comment (shiftinv) [reply disnake/ext/commands/params.py:715]: When `max_value > 2**53` but `large` is set to false, for example.
I suppose the implicit behavior is *fine* as well, I'd just like to avoid situations where someone sets a high `max_value` and wonders why they can enter letters in their slash command, since `large` just uses a string parameter under the hood.

[PEP 637](https://peps.python.org/pep-0637/) could've been nice, since it would've allowed something like `Range[5, 10, large=True]`, but that was rejected.
One possible solution for having users explicitly specify the behavior they want would be a `LargeRange` type; seems a bit clunky, but I can't think of a better solution right now (@onerandomusername ideas?).

**Comment by onerandomusername** [comment]
After reading the comments that @shiftinv has left, I think the best course of action is to leave Range as it currently is. Instead, add a new class called LargeRange (name can be bikeshedded), which switches between the two if needed. In addition, this is large by default, but can switch to float or int if the max is below the threshold.

LargeFloat would not be needed, as it would be a part of LargeRange (again, bikeshed name)

So to summarize, my proposed changes are:
- keep existing implementation of Range the same
- make LargeRange function like existing Range class, but with automatic large support and default to large.
- remove LargeFloat
- create and add any necessary conversion errors

File: disnake/ext/commands/params.py [hunk 3]
```diff
@@ -615,11 +621,35 @@ async def verify_type(self, inter: ApplicationCommandInteraction, argument: Any)
...
+                try:
+                    argument = int(argument)
+                except ValueError:
+                    raise errors.LargeIntConversionFailure(argument) from None
+            elif self._original_large_type == float:
+                try:
+                    argument = float(argument)
+                    print(argument)
+                except ValueError:
+                    raise errors.LargeFloatConversionFailure(argument) from None
```
[the remark this answers is not in the document]
Comment (EmreTech) [reply disnake/ext/commands/params.py:641]: 0fa5e2a

File: disnake/ext/commands/params.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (EmreTech) [reply disnake/ext/commands/params.py]: 0fa5e2a

**Comment by shiftinv** [comment]
As of pyright 1.1.299, using non-generic metaclasses with `__getitem__` in type expressions no longer works. This means most of `Range` and `String` will have to be rewritten to make use of `typing.Annotated` in some way.
I'll close this PR for the time being, since this will likely end up being added as part of said rewrite; thank you for the contribution though :)
